### ISSUE TITLE
No response (White Screen) when generating Scaffold Webtools

### ISSUE THREAD

**Issue by luisfelipeharo** [phalcon/phalcon-devtools#1320]
Trying to generate a Scaffold and I get no response & white screen.

https://ibb.co/vxDwnGb

Added ('staticUri'      => '/CRUD/',) and changed 'baseUri'        => '/CRUD',  as CRUD is the project name

### Details

Phalcon DevTools (3.4.0)

Environment:
  OS: Windows NT FELIPE 10.0 build 18362 (Windows 10) AMD64
  PHP Version: 7.1.32
  PHP SAPI: cli
  PHP Bin: D:\Xampp\php\php.exe
  PHP Extension Dir: C:\php\ext
  PHP Bin Dir: C:\php
  Loaded PHP config: D:\Xampp\php\php.ini
Versions:
  Phalcon DevTools Version: 3.4.0
  Phalcon Version: 3.4.4
  AdminLTE Version: 2.3.6

phalcon

Web framework delivered as a C-extension for PHP
phalcon => enabled
Author => Phalcon Team and contributors
Version => 3.4.4
Build Date => Jun 30 2019 15:35:58
Powered by Zephir => Version 0.10.15

Directive => Local Value => Master Value
phalcon.db.escape_identifiers => On => On
phalcon.db.force_casting => Off => Off
phalcon.orm.events => On => On
phalcon.orm.virtual_foreign_keys => On => On
phalcon.orm.column_renaming => On => On
phalcon.orm.not_null_validations => On => On
phalcon.orm.exception_on_failed_save => Off => Off
phalcon.orm.enable_literals => On => On
phalcon.orm.late_state_binding => Off => Off
phalcon.orm.enable_implicit_joins => On => On
phalcon.orm.cast_on_hydrate => Off => Off
phalcon.orm.ignore_unknown_columns => Off => Off
phalcon.orm.update_snapshot_on_save => On => On
phalcon.orm.disable_assign_setters => Off => Off

PHP Version 

PHP 7.1.32 (cli) (built: Aug 28 2019 09:09:30) ( ZTS MSVC14 (Visual C++ 2015) x64 )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.1.0, Copyright (c) 1998-2018 Zend Technologies

Operating System: Windows 10 Home Version: 1903

Server: Apache (Xampp).
Database: MySQL.

*Google Captcha on the forum on working. https://ibb.co/pxP5M5q

**Comment by Jeckerson** [comment]
Check your web server logs. Maybe there are some errors.

**Comment by Ruzgfpegk** [comment]
Same here.

No error, just:
`"POST /webtools.php?_url=/scaffold/generate HTTP/1.1" 200 -`

Win10 1903 / XAMPP 7.3.9
PHP Version 7.3.9
Phalcon DevTools Version 3.4.0
Phalcon Version 3.4.4

So maybe this has to do with a default setting of XAMPP's PHP7.
I'll let you know if I find something.

I had no issue generating models though.

**Comment by Jeckerson** [comment]
@luisfelipeharo @Ruzgfpegk Can you provide your config file?

**Comment by luisfelipeharo** [comment]
Sure, this is my config file.

[config.zip](https://github.com/phalcon/phalcon-devtools/files/3709676/config.zip)

**Comment by Ruzgfpegk** [comment]
Alright.

[config.zip](https://github.com/phalcon/phalcon-devtools/files/3710617/config.zip)

The structure of this->option in [Scaffold.php](https://github.com/phalcon/phalcon-devtools/blob/3.4.x/scripts/Phalcon/Builder/Scaffold.php) just before the error on line 91 is as follows:
```
Phalcon\Builder\Options::__set_state(array(
   '_path_delimiter' => NULL,
   'name' => 'languages',
   'schema' => 'project',
   'force' => NULL,
   'genSettersGetters' => '1',
   'directory' => 'C:\\Sources\\project\\webapp',
   'templatePath' => 'C:\\Sources\\project\\vendor\\phalcon\\devtools\\templates',
   'templateEngine' => 'volt',
   'modelsNamespace' => '',
))
```
No trace of 'config' there.

I may be suggesting a completely wrong fix as I'm not familiar with the Phalcon code, but in the [ScaffoldController](https://github.com/phalcon/phalcon-devtools/blob/3.4.x/scripts/Phalcon/Web/Tools/Controllers/ScaffoldController.php) (lines 60+) we see that the Scaffold is initialized in the method generateAction without the "config" parameter, while the [ModelsController](https://github.com/phalcon/phalcon-devtools/blob/3.4.x/scripts/Phalcon/Web/Tools/Controllers/ModelsController.php) (lines 228+) which worked correctly includes said parameter, so after adding the relevant line the generation works.

> Scaffold for table "_projectTable_" was generated successfully

The files for the view and controller of the tested table seem to have been correctly initialized too.

**Comment by luisfelipeharo** [comment]
> Alright.
> 
> [config.zip](https://github.com/phalcon/phalcon-devtools/files/3710617/config.zip)
> 
> The structure of this->option in [Scaffold.php](https://github.com/phalcon/phalcon-devtools/blob/3.4.x/scripts/Phalcon/Builder/Scaffold.php) just before the error on line 91 is as follows:
> 
> ```
> Phalcon\Builder\Options::__set_state(array(
>    '_path_delimiter' => NULL,
>    'name' => 'projectTable',
>    'schema' => 'project',
>    'force' => NULL,
>    'genSettersGetters' => '1',
>    'directory' => 'C:\\Sources\\project\\webapp',
>    'templatePath' => 'C:\\Sources\\project\\vendor\\phalcon\\devtools\\templates',
>    'templateEngine' => 'volt',
>    'modelsNamespace' => '',
> ))
> ```
> 
> No trace of 'config' there.
> 
> I may be suggesting a completely wrong fix as I'm not familiar with the Phalcon code, but in the [ScaffoldController](https://github.com/phalcon/phalcon-devtools/blob/3.4.x/scripts/Phalcon/Web/Tools/Controllers/ScaffoldController.php) (lines 60+) we see that the Scaffold is initialized in the method generateAction without the "config" parameter, while the [ModelsController](https://github.com/phalcon/phalcon-devtools/blob/3.4.x/scripts/Phalcon/Web/Tools/Controllers/ModelsController.php) (lines 228+) which worked correctly includes said parameter, so after adding the relevant line in ScaffoldController (line 69: +`'config' => $this->config,`) the generation works.
> 
> > Scaffold for table "_projectTable_" was generated successfully
> 
> The files for the view and controller of the tested table seem to have been correctly initialized too.

Tried it, but not working for me.

**Comment by Ruzgfpegk** [comment]
After patching the class I also got the blank screen if I forgot to put the namespace or if files from a previous run were still present.

**Comment by Jeckerson** [comment]
Released new version [v3.4.4](https://github.com/phalcon/phalcon-devtools/releases/tag/v3.4.4)
Try to update and check if now it works.

Closing for now, feel free to re-open or leave a comment, if problem continues.

**Comment by luisfelipeharo** [comment]
Still not working, got this in error log.

[php7:error] [pid 15264:tid 1984] [client ::1:52215] PHP Fatal error:  Uncaught Error: Access to undeclared static property: Phalcon\\Di::$_default in D:\\Xampp\\phalcon-devtools\\scripts\\Phalcon\\Bootstrap.php:150\nStack trace:\n#0 [internal function]: Phalcon\\Di->__construct()\n#1 D:\\Xampp\\phalcon-devtools\\scripts\\Phalcon\\Bootstrap.php(150): Phalcon\\Di\\FactoryDefault->__construct()\n#2 D:\\Xampp\\htdocs\\CRUD\\public\\webtools.php(30): Phalcon\\Bootstrap->__construct(Array)\n#3 {main}\n\nNext Error: Access to undeclared static property: Phalcon\\Di::$_default in D:\\Xampp\\phalcon-devtools\\scripts\\Phalcon\\Bootstrap.php:150\nStack trace:\n#0 [internal function]: Phalcon\\Di->__construct()\n#1 D:\\Xampp\\phalcon-devtools\\scripts\\Phalcon\\Bootstrap.php(150): Phalcon\\Di\\FactoryDefault->__construct()\n#2 D:\\Xampp\\htdocs\\CRUD\\public\\webtools.php(30): Phalcon\\Bootstrap->__construct(Array)\n#3 {main}\n  thrown in D:\\Xampp\\phalcon-devtools\\scripts\\Phalcon\\Bootstrap.php on line 150

**Comment by Jeckerson** [comment]
@luisfelipeharo 
Current problem is related with Phalcon on Windows and PHP versions below `<7.2`.
Can you upgrade your PHP version to `7.2` or `7.3` ?

Reference: https://github.com/phalcon/cphalcon/issues/12056

**Comment by sergeyklay** [comment]
Or use Nginx

**Comment by Ruzgfpegk** [comment]
For me it's the same as before after updating through composer, I had to add `'config'            => $this->config,` to ScaffoldController.php (line 69) for the scaffolding process to work through the webtools, else I got a blank screen.

Phalcon DevTools Version 	3.4.5
Phalcon Version 	3.4.4
AdminLTE Version 	2.3.6

**Comment by Jeckerson** [comment]
Further fixes will be done into `4.0.0` branch.

**Comment by Jeckerson** [comment]
Fixed in `v3.4.8` release and in `4.0.x` branch. 

Feel free to re-open if problem continues.
